src/addBusinessDays/index.ts

```diff
@@ -118,29 +131,34 @@ export default function addBusinessDays(
           return businessDaysDelta
       }
     }, 0) - accountedForExceptions
+
   // Add or subtract days until we have applied all our exceptions
   while (dayChangesDueToExceptions !== 0) {
     const deltaSign = dayChangesDueToExceptions < 0 ? -1 : 1
-    if (businessDays.includes(date.getDay())) {
+    if (isWorkingDay(newDate) || isExcepted(newDate) === false) {
       dayChangesDueToExceptions = dayChangesDueToExceptions - deltaSign
+      // don't add to newDate if the last day is a false exception
+      if (dayChangesDueToExceptions === 0 && isExcepted(newDate) === false)
+        break
     }
-    date.setDate(date.getDate() + deltaSign)
+    newDate.setDate(newDate.getDate() + deltaSign)
   }
 
   // If we land on a non-working date, we add days accordingly to land on the next business day
   const reduceIfNonWorkingDay = (date: Date) => {
-    if (isNonWorkingDay(date) && amount !== 0) {
+    if (!isWorkingDay(date) && amount !== 0) {
       const newSign = startedOnNonWorkingDay ? -sign : sign
```

Comment:
@chr1sjf0x do you have any insight into why the `newSign` depends on whether `startedOnNonWorkingDay` is true? I have a failing test that starts and ends on excluded exceptions and this `newSign` ends up changing dates in the wrong direction, but if I try to just use `sign` then I have other tests that fail.

Reply:
In one of the cases that Khalid fixed the other day, the behaviour of using sign as is resulted in too few business days being added when starting on a non-working day. This switch got it working, but at the expense of your new cases 🤔